docs/source/task_guides/ptuning-seq-classification.mdx

```diff
@@ -0,0 +1,237 @@
+
+tokenizer = AutoTokenizer.from_pretrained(model_name_or_path, padding_side=padding_side)
+if getattr(tokenizer, "pad_token_id") is None:
+    tokenizer.pad_token_id = tokenizer.eos_token_id
+
+
+def tokenize_function(examples):
+    # max_length=None => use the model max length (it's actually the default)
+    outputs = tokenizer(examples["sentence1"], examples["sentence2"], truncation=True, max_length=None)
+    return outputs
+```
+
+Use [`~datasets.Dataset.map`] to apply the `tokenize_function` to the dataset, and remove the unprocessed columns because the model won't need those. You should also rename the `label` column to `labels` because that is the expected name for the labels by models in the 🤗 Transformers library.
+
+```py
+tokenized_datasets = datasets.map(
+    tokenize_function,
+    batched=True,
+    remove_columns=["idx", "sentence1", "sentence2"],
+)
+
+tokenized_datasets = tokenized_datasets.rename_column("label", "labels")
+```
+
+Create a collator function to pad the examples in the batches to the `longest` sequence in the batch, and create a [`DataLoader`](https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader) from the `train` and `validation` splits:
+
+```py
+def collate_fn(examples):
+    return tokenizer.pad(examples, padding="longest", return_tensors="pt")
+
+
+train_dataloader = DataLoader(tokenized_datasets["train"], shuffle=True, collate_fn=collate_fn, batch_size=batch_size)
+eval_dataloader = DataLoader(
+    tokenized_datasets["validation"], shuffle=False, collate_fn=collate_fn, batch_size=batch_size
+)
+```
+
+## Train
+
+Let's create the [`PeftModel`] now. P-tuning uses a prompt encoder to optimize the prompt parameters, so you'll need to initialize the [`PromptEncoderConfig`] and define the task type:
+
+```py
+peft_config = PromptEncoderConfig(task_type="SEQ_CLS", num_virtual_tokens=20, encoder_hidden_size=128)
+```
+
+Create the base `roberta-large` model from [`~transformers.AutoModelForSequenceClassification`], and then wrap the base model and `peft_config` with [`get_peft_model`] to create a [`PeftModel`]. If you're curious to see how many parameters you're actually training, you can print it out with [`~peft.PeftModel.print_trainable_parameters`]:
+
+```py
+model = AutoModelForSequenceClassification.from_pretrained(model_name_or_path, return_dict=True)
+model = get_peft_model(model, peft_config)
+model.print_trainable_parameters()
+"trainable params: 1351938 || all params: 355662082 || trainable%: 0.38011867680626127"
+```
+
+Setup an optimizer and learning rate scheduler:
+
+```py
+optimizer = AdamW(params=model.parameters(), lr=lr)
+
+lr_scheduler = get_linear_schedule_with_warmup(
+    optimizer=optimizer,
+    num_warmup_steps=0,
+    num_training_steps=(len(train_dataloader) * num_epochs),
+)
+```
+
+Move the model to the GPU, and then write a training loop that includes the `metric` you loaded earlier to evaluate the model at each epoch:
+
+```py
+model.to(device)
```

Comment:
Is there a reason why we don't use Trainer from Transformers here?

Reply:
Hello @stevhliu  and @MKhalusova, Trainer and Accelerate are closely supported by PEFT. 

All the example scripts use Accelerate, for example, https://github.com/huggingface/peft/blob/main/examples/conditional_generation/peft_lora_seq2seq_accelerate_ds_zero3_offload.py and https://github.com/huggingface/peft/blob/main/examples/causal_language_modeling/peft_lora_clm_accelerate_ds_zero3_offload.py. 

All the INT8 quantization examples mostly use Trainer, for example, https://github.com/huggingface/peft/blob/main/examples/int8_training/Finetune_flan_t5_large_bnb_peft.ipynb and https://github.com/huggingface/peft/blob/main/examples/int8_training/peft_bnb_whisper_large_v2_training.ipynb

Sequence Classification examples are simple and I habitually use native PyTorch in notebooks as I mostly write scripts with Accelerate which closely follows native PyTorch. Feel free to switch to Trainer/Accelerate notebook launcher for this example.